main.tf

```
@@ -99,3 +99,36 @@ resource "aws_guardduty_publishing_destination" "this" {
   kms_key_arn      = var.publishing_destination.kms_key_arn
   destination_type = "S3" # S3 is currently the only option for this
 }
+
+# Provides a resource to manage a GuardDuty malware protection plan.
+resource "aws_guardduty_malware_protection_plan" "this" {
+  for_each = var.protection_plan
+
+  role = each.value.role
+
+  dynamic "protected_resource" {
+    for_each = [each.value.protected_resources]
+
+    content {
+      s3_bucket {
+        bucket_name = protected_resource.value.s3_bucket.bucket_name
+
+        object_prefixes = lookup(protected_resource.value.s3_bucket, "object_prefixes", null)
```

No need for the lookup, since the optional object-type does this for us.


```suggestion
        object_prefixes = each.value.protected_resource.s3_bucket.object_prefixes
```